clients/src/main/java/org/apache/kafka/clients/NetworkClient.java

```diff
@@ -915,6 +915,14 @@ public void handleAuthenticationFailure(AuthenticationException exception) {
         public void handleCompletedMetadataResponse(RequestHeader requestHeader, long now, MetadataResponse response) {
             this.metadataFetchInProgress = false;
             Cluster cluster = response.cluster();
+
+            long missingListenerCount = response.topicMetadata().stream()
+                    .flatMap(topicMetadata -> topicMetadata.partitionMetadata().stream())
+                    .filter(partitionMetadata -> partitionMetadata.error() == Errors.LISTENER_NOT_FOUND_ON_LEADER)
+                    .count();
+            if (missingListenerCount != 0)
+                log.error("{} partitions have leader brokers without a matching listener ", missingListenerCount);
```

Comment:
Would it be useful to have the partitions or a subset of them instead of just the count?

Reply:
@ijuma  Thanks for the review, updated.